shub_image/list.py

```diff
@@ -0,0 +1,124 @@
+import os
+import re
+import click
+import requests
+from urlparse import urljoin
+
+from shub import exceptions as shub_exceptions
+from shub.deploy import list_targets
+from shub_image import utils
+
+
+SETTING_TYPES = ['project_settings',
+                 'organization_settings',
+                 'enabled_addons']
+
+SHORT_HELP = 'List spiders.'
+
+HELP = """
+List command tries to run your image locally and get a spiders list.
+
+Internally, this command is a simple wrapper to `docker run` and uses
+docker daemon on your system to run a new container using your image.
+Before creating the container, there's a Dash call to get your project
+settings to get your spiders list properly (respecting SPIDERS_MODULE
+setting, etc).
+
+Image should be set via scrapinghub.yml, section "images". If version is not
+provided, the tool uses VCS-based stamp over project directory (the same as
+shub utils itself).
+"""
+
+
+@click.command(help=HELP, short_help=SHORT_HELP)
+@click.argument("target", required=False, default="default")
+@click.option("-l", "--list-targets", help="list available targets",
+              is_flag=True, is_eager=True, expose_value=False,
+              callback=list_targets)
+@click.option("-d", "--debug", help="debug mode", is_flag=True)
+@click.option("--version", help="release version")
+def cli(target, debug, version):
+    list_cmd_full(target, debug, version)
+
+
+def list_cmd_full(target, debug, version):
+    config = utils.load_release_config()
+    image = config.get_image(target)
+    version = version or config.get_version()
+    image_name = utils.format_image_name(image, version)
+
+    # FIXME there can be a case when there's no project specified
+    # for the target, in this case we should notify about it and
+    # skip getting settings from Dash.
```

Comment:
looks like this can be removed - I see you added a warning here https://github.com/scrapinghub/shub-image/pull/8/files#diff-8d596a8abae433cf93033ece8f474895R59